app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -78,27 +81,61 @@ def validate_missing_fields
     required.each_with_object([]) do |field, invalid_fields|
       next if @product[field.to_sym].present?
 
-      invalid_fields << {
-        field: field,
-        text: t("coronavirus_form.questions.#{controller_name}.#{field}.custom_error",
-                default: t("coronavirus_form.errors.missing_mandatory_text_field",
-                           field: t("coronavirus_form.questions.#{controller_name}.#{field}.label")).humanize),
-      }
+      invalid_fields << error_response(field)
     end
   end
 
+  def validate_product_quantity
+    field = "product_quantity"
+    quantity = Integer(@product[field.to_sym])
+    return if quantity && quantity.positive?
+
+    error_response(field)
+  rescue TypeError, ArgumentError
```

Comment:
This is reasonable 👍I wonder if it might hide genuine bugs from Sentry. Would it be possible to guard against these errors with e.g. `return error_response(field) unless @product[field.to_sym]`?

Reply:
👍 Doing that would remove the need to catch the TypeError.